### ISSUE TITLE
Add ability for branded APKs to deep link/register intents to wildcard domains

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-android#310]
**What feature do you want to improve?**
Right now we tightly bind the APKs intent to the `app_host` (see [step 3 in "New Brand"](https://docs.communityhealthtoolkit.org/apps/guides/android/branding/#2-new-brand)). This does not enable a way to link to something like `*.cht.domain.com`.  

**Describe the improvement you'd like**
Enable an extensible way to add a wildcard domain for all intents.  

**Describe alternatives you've considered**
NA

**Additional context**
This will work well when we deliver on #309.  We also already [have a deployment that needs this](https://github.com/medic/cht-android/releases/tag/untagged-fb666090af5fac8eb5b5).

**Comment by jkuester** [comment]
@mrjones-plip @kennsippell I am slightly concerned that wildcards in the intent might be more trouble than it is work in the long term (e.g. when we go to fix #308).  Auto-verifying a wildcard intent in Android 12+ [is going to _require_](https://developer.android.com/training/app-links/verify-android-applinks#multi-subdomain) the `assetlinks.json` file be published _"at the root hostname"._  What is tricky here is that in most of these cases I can think of where the wildcard intent would be useful, there will probably not actually be a CHT instance running on the root domain. So, the deployment of that file would have to be a custom/bespoke process for each case....

The Android docs I linked note that registering _multiple links for the intent_ works different from a wildcard intent.  When multiple links are registered, the `assetlinks.json` file needs to be accessible at each of the links.  Technically this will be more moving parts when managing multiple CHT instances because you will end up with `n` files that have to be deployed.  However, a likely solution to #308 is to add support to cht-core/cht-conf for just deploying the file to the CHT server along with the rest of the configuration....

Also, for what it is worth, additional custom links can already be registered by specific brands just by creating a custom `AndroidManifest.xml` file for the brand and adding something like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:tools="http://schemas.android.com/tools"
  package="org.medicmobile.webapp.mobile">

  <application>
    <activity android:name="AppUrlIntentActivity"
      android:launchMode="singleInstance"
      android:exported="true">
      <intent-filter>
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="@string/scheme" android:host="not.app_host.com" android:pathPattern=".*"/>
      </intent-filter>
    </activity>
  </application>
</manifest>
```

When the brand gets built, this AndroidManifest gets merged with the `main` one.

**Comment by mrjones-plip** [comment]
Thanks for the update @jkuester !  

tl;dr - feel free to make this work at scale using the best means possible.  I think using multiple links might be more scalable than a wildcard per you above concerns.  My detailed thoughts below.

cc @kennsippell and @dianabarsan and @Hareet 

----

I think it is fair to make this somewhat difficult, but possible, for a national scale deployment with multiple CHT instances, but easy for a single CHT instance. 

> When multiple links are registered, the `assetlinks.json` file needs to be accessible at each of the links.

I believe this means we can fix #308, again presuming CHT Core/`cht-conf` allow the instance to natively host the `assetlinks.json` file, by publishing as many URLs in Manifest [as shown in the docs](https://developer.android.com/training/app-links/verify-android-applinks#multi-host) instead of a wildcard. Here's a handy inline example illustrating the `.net` and `.com` intents in the same manifest:

```xml
<application>

  <activity android:name="MainActivity">
    <intent-filter android:autoVerify="true">
      <action android:name="android.intent.action.VIEW" />
      <category android:name="android.intent.category.DEFAULT" />
      <category android:name="android.intent.category.BROWSABLE" />
      <data android:scheme="http" />
      <data android:scheme="https" />
      <data android:host="www.example.com" />
    </intent-filter>
  </activity>
  <activity android:name="SecondActivity">
    <intent-filter>
      <action android:name="android.intent.action.VIEW" />
      <category android:name="android.intent.category.DEFAULT" />
      <category android:name="android.intent.category.BROWSABLE" />
      <data android:scheme="https" />
     <data android:host="www.example.net" />
    </intent-filter>
  </activity>

</application>
```

On the server side of the equation it's a huge benefit to use a single APK, like in #309, as it means there's just a single `assetlinks.json`  file with a shared `package_name` and  `sha256_cert_fingerprints` for all servers visible in the list at launch. [See docs](https://developer.android.com/training/app-links/verify-android-applinks#web-assoc) for more info on these two tokens used in the JSON.

**Comment by kennsippell** [comment]
I've updated https://github.com/medic/cht-android/pull/311 to take this approach.
